### PR TITLE
Remove unused code (4/4) 

### PR DESCRIPTION
**Pull request by est31**
Fourth and final PR in a series of PRs to remove unused/dead code from servo, powered by an (upcoming) tool of mine. Please take a look and tell me if you want to keep something.

* First PR: #23477
* Second PR: #23498
* Third PR: #23499

Shortstat of the combined PR series:
```
47 files changed, 7 insertions(+), 805 deletions(-)
```

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because they only remove dead code

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23532)

### CODE REVIEW

File: components/selectors/matching.rs [hunk 1]
```diff
@@ -13,11 +13,6 @@ use std::iter;
 
 pub use crate::context::*;
 
-// The bloom filter for descendant CSS selectors will have a <1% false
-// positive rate until it has this many selectors in it, then it will
-// rapidly increase.
-pub static RECOMMENDED_SELECTOR_BLOOM_FILTER_SIZE: usize = 4096;
```
Comment (emilio) [inline components/selectors/matching.rs]: This is part of the public API even if not used in Servo directly.

File: components/style/logical_geometry.rs [hunk 2]
```diff
@@ -123,11 +123,6 @@ impl WritingMode {
         !self.intersects(WritingMode::RTL)
     }
 
-    #[inline]
-    pub fn is_sideways(&self) -> bool {
```
Comment (emilio) [inline components/style/logical_geometry.rs]: This is probably still useful, even if servo doesn't support sideways writing modes yet.

**Review by emilio — APPROVED** [review]
Thanks!

**Comment by est31** [comment]
re-added the parts. r? @emilio

**Comment by jdm** [comment]
@bors-servo retry
* #23290

**Pull request merged**